src/loader/reduce_footpaths.cc

```diff
@@ -0,0 +1,107 @@
+#include "nigiri/loader/reduce_footpaths.h"
+
+#include "utl/erase_duplicates.h"
+#include "utl/insert_sorted.h"
+#include "utl/parallel_for.h"
+#include "utl/timer.h"
+
+#include "nigiri/timetable.h"
+
+namespace nigiri::loader {
+
+constexpr auto const kN = 2U;
+
+vecvec<location_idx_t, footpath> reduce_footpaths(
+    timetable& tt, vecvec<location_idx_t, footpath> const& fps) {
+  auto const timer = utl::scoped_timer{"reduce-footpaths"};
+
+  auto const init = []() {
+    auto x = std::array<footpath, kN>{};
+    x.fill(footpath{footpath::kMaxTarget, footpath::kMaxDuration});
+    return x;
+  }();
+
+  auto reduced =
+      vector_map<location_idx_t, std::vector<footpath>>(tt.n_locations());
+
+  struct state {
+    vector_map<route_idx_t, std::array<footpath, kN>> route_fps_;
+    bitvec_map<route_idx_t> route_is_reachable_;
+  };
+
+  utl::parallel_for_run_threadlocal<state>(
+      tt.n_locations(), [&](state& s, std::size_t const l_idx) {
+        auto const l = location_idx_t{l_idx};
+
+        if (s.route_fps_.size() != tt.n_locations()) {
+          s.route_fps_ = vector_map<route_idx_t, std::array<footpath, kN>>{
+              tt.n_locations(), init};
+          s.route_is_reachable_.resize(tt.n_locations());
+        }
+
+        // Group by route (duplicates footpaths).
+        // This eliminates footpaths to locations w/o scheduled traffic.
```

Comment:
How about footpaths to locations which doesn't have scheduled traffic, but are intended to be used for real-time trips, for example, stops reserved by rail replacement buses or spare rail platforms?

Reply:
They will be recovered from `footpaths_full_out_` which stores the full set of footpaths (unfiltered) and have to be copied to the `rt_timetable` in case a previously unused stop will be visited.

The main goal of this branch was to figure out the speedup potential. Results so far:

- negligible speedup for default footpaths
- measurable speedup for 20min walking footpaths
- to be measured for longer settings (probably speedup will be higher the greater the distance, but still calculation time will grow in total)